.github/workflows/Pitests.yml

```diff
@@ -0,0 +1,639 @@
+      - name: Setup local maven cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Pitest-ant
+        run: |
+          mkdir pitest-ant-report
+          ./.ci/pitest.sh pitest-ant
+      - name: Copy result
+        if: always()
+        run: cp -r target/pit-reports/ pitest-ant-report/
+      - name: Archive code coverage results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: pitest1-ant-coverage-report
+          path: /home/runner/work/checkstyle/checkstyle/pitest-ant-report/
+          retention-days: 3
+
+
+  Pitest-annotation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup local maven cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Pitest-annotation
+        run: |
+          mkdir pitest-annotation-report
+          ./.ci/pitest.sh pitest-annotation
+      - name: Copy result
+        if: always()
+        run: cp -r target/pit-reports/ pitest-annotation-report/
+      - name: Archive code coverage results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: pitest4-annotation-coverage-report
+          path: /home/runner/work/checkstyle/checkstyle/pitest-annotation-report/
+          retention-days: 3
+
+  Pitest-whitespace:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup local maven cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Pitest-whitespace
+        run: |
+          mkdir pitest-whitespace-report
+          ./.ci/pitest.sh pitest-whitespace
+      - name: Copy result
+        if: always()
+        run: cp -r target/pit-reports/ pitest-whitespace-report/
+      - name: Archive code coverage results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: pitest4-whitespace-coverage-report
+          path: /home/runner/work/checkstyle/checkstyle/pitest-whitespace-report/
+          retention-days: 3
+
+  Pitest-filters:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup local maven cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Pitest-filters
+        run: |
+          mkdir pitest-filters-report
+          ./.ci/pitest.sh pitest-filters
+      - name: Copy result
+        if: always()
+        run: cp -r target/pit-reports/ pitest-filters-report/
```

Comment:
Why not just upload the report from `/target/pit-reports/` instead of copying?

Reply:
So that content inside every folder doesn't be pit-report but of which pitest name. Removed now.

Reply:
Please help with https://github.com/checkstyle/checkstyle/pull/10379#discussion_r674531564 . Need a better solution.